lib/vrl/tests/tests/expressions/if_statement/multiline_predicates.vrl

```diff
@@ -0,0 +1,42 @@
+# result: [true, true, true, true, true]
+
+x = 2
+
+v1 = if (x = x + 1; x == 3) {
+  true
+}
+
+v2 = if (x = x + 1
+         x == 4) {
+  true
+}
+
+v3 = if (
+  x = x + 1
+  x == 5
+) {
+  true
+}
+
+v4 = if (
+  # Lets test spaces
+  x = x + 1
+
+  # Does it work?
+
+  x = x + 1; x == 7;
```

Comment:
This final `;` shouldn't be needed.

Reply:
In fact, we should disallow it, as it means the `x == 7` expression shouldn't be used as the resulting expression for the predicate. Instead, the predicate should resolve to `null`, given that there isn't any expression after `;`.

Reply:
As discussed, I am finding this to be quite a tricky thing to do, it brings up all sorts of ambiguity issues in lalrpop. I have tested on master and this is an issue there as well, so this is not a regression. I would like to defer this and have raised #9255 to track this.